File: code/ml/data_ingestion.py
from os import environ

from boto3 import client


s3_endpoint_url = environ.get('S3_ENDPOINT')
s3_access_key = environ.get('S3_ACCESS_KEY_ID')
s3_secret_key = environ.get('S3_SECRET_ACCESS_KEY')
s3_bucket_name = environ.get('S3_BUCKET')


print(f'Downloading data "raw-data.csv" '
      f'from bucket "{s3_bucket_name}" '
      f'from S3 storage at {s3_endpoint_url}')

s3_client = client(
    's3', endpoint_url=s3_endpoint_url,
    aws_access_key_id=s3_access_key, aws_secret_access_key=s3_secret_key
)

s3_client.download_file(
    s3_bucket_name,
    'data/raw-data.csv',
    'raw_data.csv'
)

File: code/ml/data_ingestion.r
library(aws.s3)


#Get environment variables to use for the connection
s3_endpoint_url <- Sys.getenv("S3_ENDPOINT")
s3_access_key <- Sys.getenv("S3_ACCESS_KEY_ID")
s3_secret_key <- Sys.getenv("S3_SECRET_ACCESS_KEY")
s3_bucket_name <- Sys.getenv("S3_BUCKET")

print(s3_endpoint_url)
#remove either HTTPS or HTTP since it doesnt want it in the URL. Obey the R-gods.
s3_endpoint_url <- sub(paste0("^", "http://"), "", s3_endpoint_url)

Sys.setenv("AWS_ACCESS_KEY_ID"=s3_access_key,
           "AWS_SECRET_ACCESS_KEY"=s3_secret_key,
           "AWS_S3_ENDPOINT"=s3_endpoint_url,
           "AWS_DEFAULT_REGION"="")

save_object("data/raw-data.csv", file = "raw_data.csv", bucket = s3_bucket_name, region ="", use_https = FALSE)

print("Download complete, LETS GO!")
